cfpb/cfgov-refresh
# Filterable List: Allow list of topics to be sorted alphabetically

In the Filterable List admin, allow the list of topics in the Topic filter to be sorted multiple ways: by frequency (current default), or alphabetically (new option).

## Additions

- New "Topic Filtering" field in Filterable List admin
- Migrations to add new field and update existing pages to use it

## Changes

- `FilterableListForm` class checks which sorting option is specified in the admin, then forms the list of options based on the selected ordering
- `filterable_list_controls` template checks the new field to determine whether to show the topic filtering control

## Testing

1. Run all migrations, then run this branch.
2. Visit any filterable list and see that topic lists are still ordered by frequency (and double-check that filters still work as intended). for example:
    - http://localhost:8000/about-us/blog/
    - http://localhost:8000/about-us/events/archive-past-events/
    - http://localhost:8000/policy-compliance/rulemaking/rules-under-development/
3. Visit the admin page for a filterable list. In the "General Content" panel, change the "Topic filtering" dropdown to "Filter topics, sort topic list alphabetically", and publish. 
4. Visit the corresponding page and see that the topic list is sorted alphabetically.

## Screenshots
![Screen Shot 2019-08-29 at 10 16 31 AM](https://user-images.githubusercontent.com/4295388/63949750-dca88b00-ca48-11e9-97e3-2a2d8db018d8.png)
![Screen Shot 2019-08-29 at 10 17 04 AM](https://user-images.githubusercontent.com/4295388/63949752-dca88b00-ca48-11e9-8bde-4830449184b2.png)
![Screen Shot 2019-08-29 at 10 17 41 AM](https://user-images.githubusercontent.com/4295388/63949753-dca88b00-ca48-11e9-8992-45b2c3557d79.png)

## Todos

After this PR is deployed to Production, I'll follow up with another PR to remove the now-outdated "Topics" checkbox from the Wagtail admin for Filterable Lists.

## Checklist
